doc/rst/technotes/firstClassProcedures.rst

```diff
@@ -0,0 +1,193 @@
+
+================================
+First-class Procedures in Chapel
+================================
+
+.. warning::
+
+  The ``lambda`` keyword (used to construct an anonymous procedure) has been
+  deprecated as of the 1.31 release. The ``func()`` procedure type constructor
+  has also been deprecated. Users should make use of the ``proc()`` syntax
+  described in this document instead.
+
+.. note::
+
+  First-class procedures are under active development and are provided
+  in their prototype form as a preview and to gather user feedback.
+
+The 1.31 release of Chapel introduced new ``proc()`` syntax for constructing
+anonymous procedures and procedure types. The next few sections provide 
+details about this syntax.
+
+The section :ref:`Capturing_First_Class_Procedures` describes when a
+procedure  may be used as a value. The restrictions in this section may
+weaken as development of first-class procedures continues.
+
+New Syntax for Constructing Procedure Types
+-------------------------------------------
+
+Procedure types may now be constructed using new syntax which mirrors
+the syntax for procedure definition.
+
+To construct the type of a procedure which takes two integers and
+returns an integer, users may write the following:
+
+.. code-block:: chapel
+
+  type T = proc(x: int, y: int): int;
+  writeln(T:string); // 'proc(x: int, y: int): int'
+
+
+.. note::
+
+  The implementation currently expects a token with the text ``proc(`` when
+  defining an anonymous procedure or procedure type. Users have expressed
+  confusion about how writing ``proc (`` (the ``proc`` token, followed by
+  one or more spaces, followed by a ``(`` token) results in a syntax error.
+  This restriction is a parsing limitation and may be removed in the future.
+
+Notice that the formals of the procedure type have specified names,
+``x`` and ``y``. Currently, formal names participate in typing such
+that a procedure defined with formals named ``x`` and ``y`` cannot be
+assigned to a procedure defined with different formal names. In the
+below example, assignment of two procedure values will fail because
+the formal names are different:
+
+.. code-block:: chapel
+
+  proc foo(x: int, y: int): int { return x + y; }
+  proc bar(a: int, b: int): int { return a + b; }
+  writeln(foo.type:string);   // 'proc(x: int, y: int): int'
+  var x = foo;                // ^ (same type) ^
+  writeln(bar.type:string);   // 'proc(a: int, b: int): int'
+  x = bar;                    // Error!
+
+There are scenarios where a user may want to pass around procedures 
+without regard for their formal names. Anonymous ``_`` formals are a
+feature under active development that are intended to support this
+behavior.
+
+New Syntax for Constructing Anonymous Procedures 
+------------------------------------------------
+
+A new syntax for constructing anonymous procedures has been introduced
+which more closely mirrors traditional procedure definition.
```

Comment:
Again, I suggest removing the above sentence "a new syntax for ..." and switch the section name to just "Anonymous Procedures."